File: riveredge-frontend/src/apps/kuaizhizao/services/purchase-order-ocr.ts
import { apiRequest, API_BASE_URL, buildAuthTenantHeaders } from '../../../services/api';

export interface PurchaseOrderOcrItem {
  materialCode?: string | null;
  materialName?: string | null;
  materialSpec?: string | null;
  materialUnit?: string | null;
  quantity?: number | null;
  unitPrice?: number | null;
  taxRate?: number | null;
  deliveryDate?: string | null;
  notes?: string | null;
}

export interface PurchaseOrderOcrResult {
  supplierName?: string | null;
  supplierContact?: string | null;
  supplierPhone?: string | null;
  orderDate?: string | null;
  deliveryDate?: string | null;
  paymentTerms?: string | null;
  currencyCode?: string | null;
  notes?: string | null;
  items: PurchaseOrderOcrItem[];
  confidenceNotes?: string | null;
}

const BASE = '/apps/kuaizhizao/purchase-orders';

export async function parsePurchaseOrderFromText(
  text: string,
  context?: PurchaseOrderOcrResult,
): Promise<PurchaseOrderOcrResult> {
  return apiRequest(`${BASE}/ocr-parse-text`, {
    method: 'POST',
    body: JSON.stringify({ text, context }),
  });
}

export async function extractPurchaseOrderFromImage(file: File): Promise<PurchaseOrderOcrResult> {
  const form = new FormData();
  form.append('file', file);
  const res = await fetch(`${API_BASE_URL}${BASE}/ocr-extract`, {
    method: 'POST',
    headers: buildAuthTenantHeaders(),
    body: form,
  });
  if (!res.ok) {
    const data = await res.json().catch(() => ({}));
    throw new Error(data?.detail?.message || data?.detail || 'OCR 识别失败');
  }
  return res.json();
}

File: riveredge-backend/src/apps/kuaizhizao/schemas/purchase_order_ocr.py
"""采购订单 OCR 智能录单 Schema"""

from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field


class PurchaseOrderOcrItem(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    material_code: Optional[str] = Field(None, alias="materialCode")
    material_name: Optional[str] = Field(None, alias="materialName")
    material_spec: Optional[str] = Field(None, alias="materialSpec")
    material_unit: Optional[str] = Field(None, alias="materialUnit")
    quantity: Optional[float] = None
    unit_price: Optional[float] = Field(None, alias="unitPrice")
    tax_rate: Optional[float] = Field(None, alias="taxRate")
    delivery_date: Optional[str] = Field(None, alias="deliveryDate")
    notes: Optional[str] = None


class PurchaseOrderOcrParseTextRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    text: str = Field(..., min_length=1)
    context: Optional["PurchaseOrderOcrResult"] = None


class PurchaseOrderOcrResult(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    supplier_name: Optional[str] = Field(None, alias="supplierName")
    supplier_contact: Optional[str] = Field(None, alias="supplierContact")
    supplier_phone: Optional[str] = Field(None, alias="supplierPhone")
    order_date: Optional[str] = Field(None, alias="orderDate")
    delivery_date: Optional[str] = Field(None, alias="deliveryDate")
    payment_terms: Optional[str] = Field(None, alias="paymentTerms")
    currency_code: Optional[str] = Field(None, alias="currencyCode")
    notes: Optional[str] = None
    items: List[PurchaseOrderOcrItem] = Field(default_factory=list)
    confidence_notes: Optional[str] = Field(None, alias="confidenceNotes")
